.github/workflows/pr-checks.yml

```
@@ -0,0 +1,107 @@
+
+jobs:
+  static-analysis:
+    name: Static Analysis
+    uses: ./.github/workflows/_static-analysis.yml
+    with:
+      go_version_file: go.work
+      golangci_lint_version: v1.64.5
+
+  unit-tests:
+    name: Unit Tests
+    if: env.SHOULD_RUN == 'true'
```

actionlint (1.7.4)

context "env" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)